src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -888,7 +925,482 @@ - (void)test005_DoBDXTransferAllowUpdateRequest
+    XCTestExpectation * applyUpdateRequestExpectation2 =
+        [self expectationWithDescription:@"handleApplyUpdateRequestForNodeID called for device2"];
+
+    XCTestExpectation * notifyUpdateAppliedExpectation1 =
+        [self expectationWithDescription:@"handleNotifyUpdateAppliedForNodeID called for device1"];
+    XCTestExpectation * notifyUpdateAppliedExpectation2 =
+        [self expectationWithDescription:@"handleNotifyUpdateAppliedForNodeID called for device2"];
+
+    __block XCTestExpectation * announceResponseExpectation2;
+
+    NSString * imagePath1 = [otaRawImagePath1 stringByReplacingOccurrencesOfString:@"raw-image" withString:@"image"];
+
+    NSString * imagePath2 = [otaRawImagePath2 stringByReplacingOccurrencesOfString:@"raw-image" withString:@"image"];
+
+    // Find the right absolute path to our ota_image_tool.py script.  PWD should
+    // point to our src/darwin/Framework, while the script is in
+    // src/app/ota_image_tool.py.
+    NSString * pwd = [[NSProcessInfo processInfo] environment][@"PWD"];
+    NSString * imageToolPath = [NSString
+        pathWithComponents:@[ [pwd substringToIndex:(pwd.length - @"darwin/Framework".length)], @"app", @"ota_image_tool.py" ]];
+
+    NSTask * task1 = [[NSTask alloc] init];
+    [task1 setLaunchPath:imageToolPath];
+    [task1 setArguments:@[
+        @"create", @"-v", @"0xFFF1", @"-p", @"0x8001", @"-vn", [kUpdatedSoftwareVersion_10 stringValue], @"-vs",
+        kUpdatedSoftwareVersionString_10, @"-da", @"sha256", otaRawImagePath1, imagePath1
+    ]];
+    NSError * launchError = nil;
+    [task1 launchAndReturnError:&launchError];
+    XCTAssertNil(launchError);
+    [task1 waitUntilExit];
+    XCTAssertEqual([task1 terminationStatus], 0);
+
+    NSTask * task2 = [[NSTask alloc] init];
+    [task2 setLaunchPath:imageToolPath];
+    [task2 setArguments:@[
+        @"create", @"-v", @"0xFFF1", @"-p", @"0x8001", @"-vn", [kUpdatedSoftwareVersion_5 stringValue], @"-vs",
+        kUpdatedSoftwareVersionString_5, @"-da", @"sha256", otaRawImagePath2, imagePath2
+    ]];
+    launchError = nil;
+    [task2 launchAndReturnError:&launchError];
+    XCTAssertNil(launchError);
+    [task2 waitUntilExit];
+    XCTAssertEqual([task2 terminationStatus], 0);
+
+    NSData * updateToken1 = [sOTAProviderDelegate generateUpdateToken];
+    NSData * updateToken2 = [sOTAProviderDelegate generateUpdateToken];
+
+    __block NSFileHandle * readHandle;
+    __block uint64_t imageSize;
+    __block uint32_t lastBlockIndex = UINT32_MAX;
+    const uint16_t busyDelay = 30; // 30 second
+    __auto_type * device1 = sConnectedDevice1;
+    __auto_type * device2 = sConnectedDevice2;
+
+    // This to keep track of whether queryImageHandler for device 2 was called or not. The first time it's called we will
+    // fulfill queryExpectation2 and start BDX for device 1.
+    static bool isQueryImageHandlerForDevice2Called = false;
```

Comment:
Why static?  I would have expected a `__block bool` for this case...  Something like:
```suggestion
    __block bool firstQueryImageForDevice2Received = false;
```